cli/src/cluster_query.rs

```diff
@@ -1094,27 +1094,45 @@ pub fn process_get_epoch(rpc_client: &RpcClient, _config: &CliConfig) -> Process
 
 pub fn process_get_epoch_info(rpc_client: &RpcClient, config: &CliConfig) -> ProcessResult {
     let epoch_info = rpc_client.get_epoch_info()?;
-    let average_slot_time_ms = rpc_client
-        .get_recent_performance_samples(Some(60))
-        .ok()
-        .and_then(|samples| {
-            let (slots, secs) = samples.iter().fold((0, 0), |(slots, secs), sample| {
-                (slots + sample.num_slots, secs + sample.sample_period_secs)
-            });
-            (secs as u64).saturating_mul(1000).checked_div(slots)
-        })
-        .unwrap_or(clock::DEFAULT_MS_PER_SLOT);
-    let start_block_time = rpc_client
-        .get_block_time(epoch_info.absolute_slot - epoch_info.slot_index)
-        .ok();
-    let current_block_time = rpc_client.get_block_time(epoch_info.absolute_slot).ok();
-    let epoch_info = CliEpochInfo {
+    let mut cli_epoch_info = CliEpochInfo {
         epoch_info,
-        average_slot_time_ms,
-        start_block_time,
-        current_block_time,
+        average_slot_time_ms: 0,
+        start_block_time: None,
+        current_block_time: None,
     };
-    Ok(config.output_format.formatted_string(&epoch_info))
+    match config.output_format {
+        OutputFormat::Json | OutputFormat::JsonCompact => {}
+        _ => {
+            let epoch_info = cli_epoch_info.epoch_info.clone();
+            let average_slot_time_ms = rpc_client
+                .get_recent_performance_samples(Some(60))
+                .ok()
+                .and_then(|samples| {
+                    let (slots, secs) = samples.iter().fold((0, 0), |(slots, secs), sample| {
+                        (slots + sample.num_slots, secs + sample.sample_period_secs)
+                    });
+                    (secs as u64).saturating_mul(1000).checked_div(slots)
+                })
+                .unwrap_or(clock::DEFAULT_MS_PER_SLOT);
+            let mut start_block_time = None;
+            let epoch_expected_start_slot = epoch_info.absolute_slot - epoch_info.slot_index;
+            let mut epoch_start_slot = epoch_expected_start_slot.saturating_sub(1);
+            while start_block_time.is_none() && epoch_start_slot <= epoch_expected_start_slot + 16 {
```

Comment:
any reason not to cheat and `getBlocksWithLimit(epoch_expected_start_slot, 1)` our way to victory?

Reply:
No indeed! That would be much better. Will rewrite.

Reply:
Well, I suppose the one possible argument against it is that get_blocks_with_limit doesn't return the block_time. So there's an extra RPC call needed in the case where the epoch doesn't start with a skipped slot. I still think it's better, though.

Reply:
@t-nelson , updated. How does that look?
I considered checking the block time of the expected epoch start slot first, and only pressing on if None, but the early block time requests are taking quite a long time against the foundation rpc cluster atm (I'm not totally sure why; need to take a deeper look) so I'm reluctant to add an extra one.